engine/class_modules/monk/sc_monk.cpp

```diff
@@ -3676,30 +3676,57 @@ struct chi_torpedo_t : public monk_spell_t
 // ==========================================================================
 // Crackling Jade Lightning
 // ==========================================================================
-// Power of the Thunder King TODO: Either
-// a) Copy Warlock's Soul Rot implementation for Drain Soul.
-// b)  channel do 0 damage and use a fake tick_action to do a single instance
-// of aoe damage on each tick, and on the tick_action override
-// action_t::amount_type() to return result_amount_type::DMG_OVER_TIME
 
 struct crackling_jade_lightning_t : public monk_spell_t
 {
+  struct crackling_jade_lightning_aoe_t : public monk_spell_t
+  {
+    crackling_jade_lightning_aoe_t( monk_t *p )
+      : monk_spell_t( p, "crackling_jade_lightning_aoe", p->baseline.monk.crackling_jade_lightning )
+    {
+      dual = background = true;
+      ww_mastery        = true;
+
+      parse_effects( p->talent.windwalker.power_of_the_thunder_king, 0B001U );
+    }
+
+    double cost_per_tick( resource_e ) const override
+    {
+      return 0.0;
+    }
+
+    double composite_persistent_multiplier( const action_state_t *action_state ) const override
+    {
+      double pm = monk_spell_t::composite_persistent_multiplier( action_state );
+
+      pm *= 1 + p()->buff.the_emperors_capacitor->check_stack_value();
+
+      return pm;
+    }
+  };
+
+  crackling_jade_lightning_aoe_t *aoe_dot;
+
   crackling_jade_lightning_t( monk_t *p, util::string_view options_str )
-    : monk_spell_t( p, "crackling_jade_lightning", p->baseline.monk.crackling_jade_lightning )
+    : monk_spell_t( p, "crackling_jade_lightning", p->baseline.monk.crackling_jade_lightning ),
+      aoe_dot( new crackling_jade_lightning_aoe_t( p ) )
   {
     sef_ability      = actions::sef_ability_e::SEF_CRACKLING_JADE_LIGHTNING;
     may_combo_strike = true;
+    ww_mastery       = true;
 
     parse_options( options_str );
 
-    channeled = tick_zero = tick_may_crit = true;
+    channeled = tick_zero = true;
     dot_duration                          = data().duration();
     interrupt_auto_attack                 = true;
     // Forcing the minimum GCD to 750 milliseconds for all 3 specs
     min_gcd  = timespan_t::from_millis( 750 );
     gcd_type = gcd_haste_type::SPELL_HASTE;
```

Comment:
It is very likely that CJL already has a hasted GCD, and does not need to be overridden.